packages/table/src/store/tree.js

```diff
@@ -59,76 +59,94 @@ export default {
 
   methods: {
     normalize(data) {
-      const { childrenColumnName, lazyColumnIdentifier, rowKey, lazy } = this.states;
+      const {
+        childrenColumnName,
+        lazyColumnIdentifier,
+        rowKey,
+        lazy
+      } = this.states;
       const res = {};
-      walkTreeNode(data, (parent, children, level) => {
-        const parentId = getRowIdentity(parent, rowKey);
-        if (Array.isArray(children)) {
-          res[parentId] = {
-            children: children.map(row => getRowIdentity(row, rowKey)),
-            level
-          };
-        } else if (lazy) {
-          // 当 children 不存在且 lazy 为 true，该节点即为懒加载的节点
-          res[parentId] = {
-            children: [],
-            lazy: true,
-            level
-          };
-        }
-      }, childrenColumnName, lazyColumnIdentifier);
+      walkTreeNode(
+        data,
+        (parent, children, level) => {
+          const parentId = getRowIdentity(parent, rowKey);
+          if (Array.isArray(children)) {
+            res[parentId] = {
+              children: children.map(row => getRowIdentity(row, rowKey)),
+              level
+            };
+          } else if (lazy) {
+            // 当 children 不存在且 lazy 为 true，该节点即为懒加载的节点
+            res[parentId] = {
+              children: [],
+              lazy: true,
+              level
+            };
+          }
+        },
+        childrenColumnName,
+        lazyColumnIdentifier
+      );
       return res;
     },
 
     updateTreeData() {
       const nested = this.normalizedData;
       const normalizedLazyNode = this.normalizedLazyNode;
       const keys = Object.keys(nested);
-      if (!keys.length) return;
-      const { treeData: oldTreeData, defaultExpandAll, expandRowKeys, lazy } = this.states;
       const newTreeData = {};
-      const rootLazyRowKeys = [];
-      const getExpanded = (oldValue, key) => {
-        const included = defaultExpandAll || (expandRowKeys && expandRowKeys.indexOf(key) !== -1);
-        return !!((oldValue && oldValue.expanded) || included);
-      };
-      // 合并 expanded 与 display，确保数据刷新后，状态不变
-      keys.forEach(key => {
-        const oldValue = oldTreeData[key];
-        const newValue = { ...nested[key] };
-        newValue.expanded = getExpanded(oldValue, key);
-        if (newValue.lazy) {
-          const { loaded = false, loading = false } = oldValue || {};
-          newValue.loaded = !!loaded;
-          newValue.loading = !!loading;
-          rootLazyRowKeys.push(key);
-        }
-        newTreeData[key] = newValue;
-      });
-      // 根据懒加载数据更新 treeData
-      const lazyKeys = Object.keys(normalizedLazyNode);
-      if (lazy && lazyKeys.length && rootLazyRowKeys.length) {
-        lazyKeys.forEach(key => {
+      if (keys.length) {
```

Comment:
@ziyoung just remove `if` statement above and add this condition to reset `states.treeData` to blank Object when there is no nested data.